source/Engine.cpp

```diff
@@ -1632,11 +1632,14 @@ void Engine::MoveShip(const shared_ptr<Ship> &ship)
 
 	// Boarding:
 	bool autoPlunder = !ship->IsYours();
-	shared_ptr<Ship> victim = ship->Board(autoPlunder);
-	if(victim)
-		eventQueue.emplace_back(ship, victim,
-			ship->GetGovernment()->IsEnemy(victim->GetGovernment()) ?
-				ShipEvent::BOARD : ShipEvent::ASSIST);
+	if(ship.get() != flagship || !ship->CanBeCarried())
```

Comment:
This makes it so that if you are in a fighter you can't board anymore. Why is this change necessary?

Reply:
The BOARD command for fighters is also used for docking. Does BOARD for a same-government ship with free bays mean; BOARD to refuel it, or BOARD to dock with it? For AI-controlled fighters we choose in the past to just don't allow boarding to resolve this ambiguity. Player controlled fighters now also follow the same behavior.

For the future I feel we should split the BOARD command in a DOCK and a BOARD command, but we are not there yet.

Reply:
That sounds reasonable, although as it currently is written it will also disallow boarding different-government ships (like a player flying a fighter that wants to board a disabled pirate, that should never be interpreted as docking though right?).

Reply:
Which is very relevant for https://github.com/endless-sky/endless-sky/pull/6323 where I probably copied this code from, but it seems less relevant for this specific PR.

I'll check if there still is a reason to have this code here as well. (And if not, then I'll remove it here.)

Reply:
I checked `Ship::Board()`, and that function is also very explicit in that ships which have `CanBeCarried()` return true can only dock and not board.
My preference would be to not spend any additional effort here to create exceptions for fighter flagships, but to address fighter boarding more genericly in some follow-up PR.

Reply:
Oh I see. The original behavior sets (for fighter flagships) CanBeCarried to false (through AllowCarried), which lets fighter flagships board (and not dock).  So that is no longer the case then. Follow PR is fine by me, but that would mean that people using fighter flagships aren't going to be able to board ships anymore until then.

Reply:
Agreed, and I already created issue https://github.com/endless-sky/endless-sky/issues/5338 to request the feature to allow (some) fighters to dock. (And to disallow docking for some other ships.)

Reply:
Just tested; the fighter does seem to start boarding, but doesn't seem to finish. There is probably still is a check somewhere that prevents boarding in this case.

Reply:
Those lines seem to prevent the boarding: https://github.com/endless-sky/endless-sky/blob/58b581f6f0ff199600e46994903cc8d812f87be3/source/Ship.cpp#L1996